### PR TITLE
Feat/user profile api

### PR DESCRIPTION
**Pull request by hieuvmo**

### CODE REVIEW

File: src/modules/Profile/components/PersonalAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    (state: RootState) => state.profile
+  );
+
+  const handleUploadImage = async (e: ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+
+    if (file) {
+      // check image size > 3MB
+      if (file.size / 1024 / 1024 > 3) {
+        notification.error({
```
Comment (Duc-Quang-0310) [inline src/modules/Profile/components/PersonalAvatar.tsx]: return notification.error

File: src/modules/Profile/components/PersonalAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          message: t("profile:update_avatar"),
+          description: t("profile:upload_size_err")
+        });
+      }
+      // check image format
+      if (!file.type.includes("image")) {
+        notification.error({
```
Comment (Duc-Quang-0310) [inline src/modules/Profile/components/PersonalAvatar.tsx]: return notification.error

File: src/redux/sagas/auth.saga.ts [hunk 3]
```diff
@@ -116,10 +131,92 @@ function* logoutActionSaga() {
...
+        message: i18nTranslate("auth:change_psw_success"),
+        success: true
+      })
+    );
+    yield put(resetAuthForm());
+    yield destroyAllLocalStorageItem();
+    yield removeAccessTokenFromCookie();
+    yield removeRefreshTokenFromCookie();
+    yield onFinish?.();
+    yield notification.success({
```
Comment (Duc-Quang-0310) [inline src/redux/sagas/auth.saga.ts]: unnecessary yield

File: src/redux/sagas/auth.saga.ts [hunk 4]
```diff
@@ -116,10 +131,92 @@ function* logoutActionSaga() {
...
+        ...response,
+        message: i18nTranslate("auth:change_psw_success"),
+        success: true
+      })
+    );
+    yield put(resetAuthForm());
+    yield destroyAllLocalStorageItem();
+    yield removeAccessTokenFromCookie();
+    yield removeRefreshTokenFromCookie();
+    yield onFinish?.();
```
Comment (Duc-Quang-0310) [inline src/redux/sagas/auth.saga.ts]: unnecessary yield

File: src/redux/sagas/auth.saga.ts [hunk 5]
```diff
@@ -116,10 +131,92 @@ function* logoutActionSaga() {
...
+      changePswActionComplete({
+        ...response,
+        message: i18nTranslate("auth:change_psw_success"),
+        success: true
+      })
+    );
+    yield put(resetAuthForm());
+    yield destroyAllLocalStorageItem();
+    yield removeAccessTokenFromCookie();
+    yield removeRefreshTokenFromCookie();
```
Comment (Duc-Quang-0310) [inline src/redux/sagas/auth.saga.ts]: unnecessary yield

Comment (Duc-Quang-0310) [inline src/redux/sagas/auth.saga.ts]: unnecessary yield

File: src/redux/sagas/auth.saga.ts [hunk 6]
```diff
@@ -116,10 +131,92 @@ function* logoutActionSaga() {
...
+    yield put(
+      updateProfileActionComplete({
+        ...response,
+        message: i18nTranslate("profile:update_profile_success"),
+        success: true
+      })
+    );
+    yield put(resetAuthForm());
+    yield onFinish?.();
+    yield notification.success({
```
Comment (Duc-Quang-0310) [inline src/redux/sagas/auth.saga.ts]: unnecessary yield

**Review by Duc-Quang-0310 — CHANGES_REQUESTED** [review]

File: src/modules/Profile/components/PersonalAvatar.tsx (hunk 1 above)
Comment (hieuvmo) [reply src/modules/Profile/components/PersonalAvatar.tsx]: why must use return ? notification is function of antd. I have tested when having error, it run good

File: src/modules/Profile/components/PersonalAvatar.tsx (hunk 2 above)
Comment (hieuvmo) [reply src/modules/Profile/components/PersonalAvatar.tsx]: why must use return ? notification is function of antd. I have tested when having error, it run good

**Pull request merged**